src/encoder/encoderrecordingsettings.h

```diff
@@ -0,0 +1,34 @@
+#pragma once
+
+#include "encoder/encodersettings.h"
+
+class EncoderRecordingSettings : public EncoderSettings {
+  public:
+    ~EncoderRecordingSettings() override = default;
+
+    // Indicates that it uses the quality slider section of the preferences
+    virtual bool usesQualitySlider() const = 0;
+    // Indicates that it uses the compression slider section of the preferences
+    virtual bool usesCompressionSlider() const = 0;
+    // Indicates that it uses the radio button section of the preferences.
+    virtual bool usesOptionGroups() const = 0;
+
+    virtual void setQualityByIndex(int qualityIndex) {
```

Comment:
Declaring the uses...() functions as pure virtual and providing empty default implementations for the setters is inconsequential. If a derived class returns true and forgets to implement the corresponding setter nothing happens.

The default implementations should at least contain a debug assert:
```
DEBUG_ASSERT(!"unimplemented");
```